omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -57,22 +57,27 @@ function onMappingSelect() {
     selectedEndDate = null;
     toggleCustomRangeCheckbox(false);
     hidePeriodPickers();
-    switch(selectedMapping.periodType){
-        case 'Daily':
-            initializeDailyPicker();
-            break;
-        case 'Weekly':
-            initializeWeeklyPicker();
-            break;
-        case 'Monthly':
-            initializeMonthlyPicker();
-            break
-        case 'Yearly':
-            initializeYearlyPicker();
-            break;
-        default:
-            jQuery('#customPeriodPicker').show();
-            toggleCustomRangeCheckbox(true);
+    // Checks whether the period type is a financial year or not
+    if (selectedMapping.periodType.split('Financial').length === 2) {
+        initializeYearlyPicker(selectedMapping.periodType.split('Financial')[1]);
```

Comment:
Financial October period type is defined as `FinancialOct` in DHIS2. Will it affect this method?

Reply:
Oh yes. Then better if we could pass the postfix to the method